docs/examples/hairpin_fitting/hairpin_unfolding.rst

```diff
@@ -0,0 +1,339 @@
+    plt.legend()
+    plt.ylabel("Force (pN)")
+    plt.subplot(2,1,2)
+    distance.plot()
+    plt.ylabel(r"Distance ($\mu$m)")
+    plt.tight_layout()
+
+.. image:: baseline_vs_time.png
+
+Below, we average the force at large distance to estimate the distance offset::
+
+    tmin_offset = "0s"
+    tmax_offset = "1s"
+
+    baseline_1x_data_hf = baseline_data["Force HF"]["Force 1x"]
+    baseline_2x_data_hf = baseline_data["Force HF"]["Force 2x"]
+
+    f1_offset = np.mean(baseline_1x_data_hf[tmin_offset:tmax_offset].data)
+    f2_offset = np.mean(baseline_2x_data_hf[tmin_offset:tmax_offset].data)
+
+    baseline_1x_no_offset = baseline_1x_data_hf - f1_offset
+    baseline_2x_no_offset = baseline_2x_data_hf - f2_offset
+
+Fit the baselines using a 7th degree polynomial function::
+
+    baseline_1x = lk.ForceBaseLine.polynomial_baseline(
+    baseline_data["Trap position"]["1X"], baseline_1x_no_offset, degree=7, downsampling_factor=500
+    )
+    baseline_2x = lk.ForceBaseLine.polynomial_baseline(
+    baseline_data["Trap position"]["1X"], baseline_2x_no_offset, degree=7, downsampling_factor=500
+    )
+
+Fit the result of the fit::
+
+    plt.figure()
+    baseline_1x.plot(label="baseline f1x")
+    baseline_2x.plot(label="baseline f2x")
+    plt.ylabel("Force (pN)")
+    plt.legend()
+
+.. image:: baselinefit.png
+
+The quality of the fit can be visualized by plotting the residuals. 
+When the `degree`` of the fitted polynomial is too low, the residuals will be large and not flat. When the degree is too high, the baseline fit above will give a warning that the fit may be poorly conditioned::
+
+    plt.figure()
+    baseline_1x.plot_residual(label="baseline f1x")
+    baseline_2x.plot_residual(label="baseline f2x")
+    plt.legend(loc='lower right')
+    plt.show()
+
+.. image:: residuals.png
+
+Compute the piezo distance
+--------------------------
+
+Now that we have determined the distance calibration and fitted the baseline, the piezo distance can be computed.
+The `signs` parameter indicates the sign of `Force 1x` and `Force 2x` respectively. By looking at the baselines, we know that `Force 1x` is negative and `Force 2x` positive::
+
+    piezo_calibration = lk.PiezoForceDistance(distance_calibration, baseline_1x, baseline_2x, signs=(-1,1)) 
+
+Choose an fd curve to compute the Piezo distance for.  
+If the force offset for the fd curve is different as compared to the offset for the baseline, it can be :ref:`included in the model <model_composition>`.
+For this experiment, the same offset that was present in the baseline, is also present in the fd curve. Therefore, we subtract it here::
+
+    fd_data = lk.File("test_data/FDCURV~3.H5")
+
+    tether_length, corrected_force_1x, corrected_force_2x = piezo_calibration.force_distance(
+    fd_data["Trap position"]["1X"], fd_data.force1x - f1_offset, fd_data.force2x - f2_offset, downsampling_factor=500
+    )
+    force_data = corrected_force_2x
+
+Plot the result::
```

Comment:
So one thing I would be asking myself as a user is, if they are roughly the same, why do we need piezo tracking for this? I get that the temporal resolution is higher, which is great when you're looking at state switching, but is that temporal resolution needed for fitting an Fd curve?

Reply:
The two reasons for using the Piezo distance is 1) higher temporal resolution and 2) for short tethers where camera tracking breaks down.  I think this should be explained in a separate document on LUMICKs University.  Indeed, for this example using Piezo distance is not really necessary.

Reply:
👍